### PR TITLE
feat: Add IStateRlayerInterface and more examples

### PR DESCRIPTION
**Pull request by cuongquangnam**
#### What this PR does / why we need it:
As per title

### CODE REVIEW

File: contracts/IStateRelayer.sol [hunk 1]
```diff
@@ -50,30 +51,34 @@ interface IStateRelayer {
...
-
-    function getBTCPrice() public returns (uint256) {
-        (, IStateRelayer.DEXInfo memory dex) = IStateRelayer(_stateRelayer).getDexPairInfo('dBTC-DFI');
-        // check if the value is maximum or not, if it is maximum, 
-        // meaning the value is invalid, revert to the previous price
-        _latestBTCPrice =  dex.primaryTokenPrice == type(uint256).max ? _latestBTCPrice : dex.primaryTokenPrice;
 
-        return _latestBTCPrice;
-    }  
 }
```
Comment (fullstackninja864) [inline contracts/IStateRelayer.sol:84]:
```suggestion
}

```

File: contracts/example/ExampleKeepValidValue.sol [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    function getBTCPrice() public returns (uint256) {
+        (, IStateRelayer.DEXInfo memory dex) = IStateRelayer(_stateRelayer).getDexPairInfo('dBTC-DFI');
+        // check if the value is maximum or not, if it is maximum, 
+        // meaning the value is invalid, revert to the previous price
+        _latestBTCPrice =  dex.primaryTokenPrice == type(uint256).max ? _latestBTCPrice : dex.primaryTokenPrice;
+
+        return _latestBTCPrice;
+    }  
+}
```
Comment (fullstackninja864) [inline contracts/example/ExampleKeepValidValue.sol:26]:
```suggestion
}

```

File: contracts/example/ExampleRevertIfInvalid.sol [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    function getBTCPrice() public view returns (uint256) {
+        (, IStateRelayer.DEXInfo memory dex) = IStateRelayer(_stateRelayer).getDexPairInfo('dBTC-DFI');
+        // check if the value is maximum or not. If it is maximum, 
+        // meaning the value is invalid, revert
+        if (dex.primaryTokenPrice == type(uint256).max) revert INVALID_DBTC_PRICE();
+
+        return dex.primaryTokenPrice;
+    }  
+}
```
Comment (fullstackninja864) [inline contracts/example/ExampleRevertIfInvalid.sol:26]:
```suggestion
}

```

File: contracts/IStateRelayer.sol (hunk 1 above)
Comment (cuongquangnam) [reply contracts/IStateRelayer.sol:84]: Why do we need to have such new lines at end of file?

Comment (fullstackninja864) [reply contracts/IStateRelayer.sol]: its coding conventions best practices of adding new line at end of each file

**Review by fullstackninja864 — APPROVED** [review]

**Pull request merged**